src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,236 @@
+        }
+
+        if(playerToTeleport == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+        teleportPlayer(sender, playerToTeleport);
+    }
+
+    /**
+     * Method used to teleport the player.
+     * @param sender the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
+        {
+            attCounter++;
+            int x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+            int z = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+
+            //Search for a ground position
+            final BlockPos groundPosition = findLand(new BlockPos(x, STARTING_Y, z), sender.getEntityWorld());
+
+            //If no position found
+            if (groundPosition == null)
+            {
+                attCounter++;
+                continue;
+            }
+
+            boolean foundPosition = isPositionSave(sender, groundPosition) && !findColony(groundPosition, sender.getEntityWorld());
+
+            /* Take the return and determine if good or bad */
+             /* send info to look to see if another colony is near */
+
+            if (foundPosition)
+            {
+                /* everything checks out good make the TP and jump out*/
+                playerToTeleport.setPositionAndUpdate(groundPosition.getX(), groundPosition.getY() + SAFETY_DROP, groundPosition.getZ());
+                return;
+            }
+        }
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Couldn't find a safe spot.  Try again in a moment."));
+    }
+
+    /**
+     * this checks that you are not in the air or underground
+     * and if so it will look up and down for a good landing spot
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @return blockPos to be used for the TP
+     */
+    private static BlockPos findLand(final BlockPos blockPos, final World world)
+    {
+        int top = STARTING_Y;
+        int bot = 0;
+        int mid = STARTING_Y;
+
+        BlockPos foundland = null;
+        BlockPos tempPos = blockPos;
+        //We are doing a binary search to limit the amount of checks (usually at most 9 this way)
+        while (top >= bot)
+        {
+            tempPos = new BlockPos( tempPos.getX(),mid, tempPos.getZ());
+            Block blocks = world.getBlockState(tempPos).getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'blocks' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)